### PR TITLE
Add script to automatically generate docs and push to gh-pages branch.

### PR DESCRIPTION
**Pull request by rcedwards**
Travis Script to:

* clone gh-pages branch
* generate new docs with Jazzy
* commit and push new docs to gh-pages

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
     on_failure: change
-    on_start: never
\ No newline at end of file
+    on_start: never
+after_success:
+- ./Resources/scripts/publish_docs.sh
+env:
+  global:
+    secure: qYJKKnzofp2TYRSR63WCzLE1yiktd13IyLXzQK8+c5h3LJYmkvAX+6XaXyCMSsQ94PQKzUrluYFHm/xfAzpbxGZmgCdmar1yD0s4NQGuQ8M0LChDTMBkyKj5IDRg3xptI9n406X28/AhkqHUtTuHG21XcMz+POYteYFq3UoeLRYbuSvOwtw+EC6cO8ANRGSINQAax0IDBvwAmN5kPdinzo9KSMOCnPT7t0Oqp4w6mB0Z/EDlJl07da+dQ4bUJGO+oOy7bgOuA+9z7MvwesniDdu5Uf/n49tBklXqQJIawQl+XY2+ehRuhDZJ0GYVSDBNJ4rWOHj7RRLucmM6H2Sm0wqXroPRf/jVzLsdrCiOLFgWlOMzej7nofCG03sK6Wfv/2n5MiXbWSfpIq9EOhPGsjgFfD36jnr49FV5HoHHrpz7bV1ebjWo75t0zUcpfzRyW9h+05JQ31gt5RwLC/cj2EV+5TRd4MXfqRrMLvj18BL8lZNXRxuOLikGombzamUUxyNH1n2QwOG5OA5XkJfhSJ+wnd20YN+gzHY8Jgjtl9CVpy3pczH4TMiQNp4QPA6EbRmdSWRtRazOF10K3QXWiZ4ZRP6uuti33heVHLXZtkBtMU7bU2/AXt1Q5A/3+ndRiWrAIxhBWrMGQP1GI00cR38uQnUcqbOrXn6yCF88+T8=
```
Comment (davidahouse) [inline .travis.yml]: I assume this is the GH_TOKEN that you referenced in the script? What account is this linked to?

File: Resources/scripts/publish_docs.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+if [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ "$TRAVIS_BRANCH" == "master" ]; then
```
Comment (davidahouse) [inline Resources/scripts/publish_docs.sh]: the above should work fine, but the double square bracket and curly brace is more bashy:

if [[ "${TRAVIS_PULL_REQUEST}" == "false" ]] && [[ "${TRAVIS_BRANCH} == "master" ]]; then

File: Resources/scripts/publish_docs.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    pushd gh-pages
+
+    echo -e "Adding new docs \n"
+    git add .
+    git commit -m "Refresh docs from successful travis build $TRAVIS_BUILD_NUMBER"
+    git push -fq origin gh-pages > /dev/null
+
+    echo -e "Moving out of gh-pages clone and cleaing up"
+    popd
+    rm -rf gh-pages
```
Comment (davidahouse) [inline Resources/scripts/publish_docs.sh]: Technically this isn't necessary. You can leave droppings in the build environment and they get reverted anyway. Also I have no idea what cleaing is... :P

File: .travis.yml [hunk 4]
```diff
@@ -1,11 +1,17 @@
...
 xcode_project: CoreDataStack.xcodeproj
 xcode_scheme: CoreDataStack
 xcode_sdk: iphonesimulator
 osx_image: xcode7.1
-before_install: brew uninstall xctool && brew install --HEAD xctool
+before_install:
+- brew uninstall xctool && brew install --HEAD xctool
 notifications:
   slack:
     secure: c8GnZ++4nvEMc9iJpKrzClvl6/rcZGjIYiVjrFnkQTxFpBeuqombiCIlwdPWFd5JxARO6h1XI6Pa/iOoOPye90/xE2w/lofoAyx+5panha8Aw+S6H2icQdez9ZatzVNzMDLtoHu1VARYCdulZa4+FvHYuMH3tkGMyekbRPgN18g9D7atd4NkmvoRAG0mNbPEiZKygqHnVRkVbWcvSsp2g80/0EedKOm01uXtkF440Y5UJ3NqC4o1dCrPXedD36H1iUtz9zKmgkJrhYsDMwA8MAwprWXM7JSRCCCjxaLK8VhsEeURzQ9VpEV1GPh8kkGsdY8uV9ET/UgOsMIlnzAtTA19XUcxtne+sDXNorGAM49MRg+fX3CefEtv1pvL8/ILM+akTyYWRubmdz9PiFa/9FWuZq5fZTwCGAJYwW7FuSYcWLnXdRDgVZzSFU0puL75INU6zwo1Jo9AVZBqULJfrNwpEy+ZkddoXQwOuX4v0l1i9RGmMQbnXEDOKZdtsLvsnuc4+9QKF1bf48bQbKy9d/zzW4X4QkLHZZpsIteQeWnwRYoFtEWseZe362PNyOIq+kAFUqhVxMlOiFczskWZ2G5uOl1mu57RboIInEf+4VdqR3SckVuNlXnK+Rsm0CFfvKj5B45QXpadukXPK/O7M1VRWQEb23HVmSxc8tA6kcc=
```
Comment (davidahouse) [inline .travis.yml]: missing the rooms: line

should be:

notifications:
  slack:
    rooms:
      - secure: ...

Comment (rcedwards) [inline .travis.yml]: Good :eyes:

File: Resources/scripts/publish_docs.sh (hunk 3 above)
Comment (rcedwards) [inline Resources/scripts/publish_docs.sh]: Sounds good I'll drop that part

File: .travis.yml (hunk 1 above)
Comment (rcedwards) [inline .travis.yml]: We can discuss this one internally.

File: Resources/scripts/publish_docs.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    git config --global user.name "travis-ci"
+    git clone --quiet --branch=gh-pages https://${GH_TOKEN}@github.com/bignerdranch/CoreDataStack gh-pages > /dev/null
+
+    echo -e "Installing Jazzy \n"
+    gem install jazzy
+
+    echo -e "Generating Jazzy output \n"
+    jazzy --swift-version 2.1 --source-directory ./ --output ./gh-pages --podspec ./BNRCoreDataStack.podspec
+
+    echo -e "Moving into gh-pages clone"
```
Comment (jeremy-w) [inline Resources/scripts/publish_docs.sh]: Unneeded `-e` flag - there are no backslash escapes here.

**Pull request merged**